xradar/accessors.py

```diff
@@ -204,3 +219,25 @@ def add_crs(self) -> dt.DataTree:
         """
         ds = self.xarray_obj
         return ds.pipe(add_crs_tree)
+
+    def apply(self, func, pass_sweep_name=False, *args, **kwargs):
+        """
+        Applies a given function to all sweeps in the radar volume.
+
+        Parameters
+        ----------
+        func : function
+            The function to apply to each sweep.
+        pass_sweep_name : bool, optional
+            Whether to pass the sweep name to the function. Defaults to False.
+        *args : tuple
+            Additional positional arguments to pass to the function.
+        **kwargs : dict
+            Additional keyword arguments to pass to the function.
+
+        Returns
+        -------
+        DataTree
+            The DataTree object after applying the function to all sweeps.
+        """
+        return apply_to_sweeps(self.xarray_obj, func, pass_sweep_name, *args, **kwargs)
```

Comment:
Let's handle pass_sweep_name via *args or **kwargs.

Reply:
I'm wondering if we could simplify this like this:

```python
radar = self.xarray_obj
for key in list(radar.children):
    if "sweep" in key:
        radar[key].ds = func(
            radar[key].to_dataset(), *args, **kwargs)
return radar
```

Reply:
Yeah, definitely we can.